src/CrossFilter.js

```diff
@@ -46,22 +47,25 @@ export default class CrossFilter {
    */
   // backwards compatibility
   dimension(expression, isGlobal = false) {
-    this._dimensions.push(expression)
-    return new Dimension(this, expression, isGlobal)
+    this._dimensionsExpessions.push(expression)
+    const dim = new Dimension(this, expression, isGlobal)
+    this._dimensions.push(dim)
+    return dim
   }
   removeDimension(dimension) {
     const { _dimensionIndex } = dimension
     this._dimensions.splice(_dimensionIndex, 1)
+    this._dimensionsExpessions.splice(_dimensionIndex, 1)
     this._filters.splice(_dimensionIndex, 1)
   }
   getDimensions() {
-    return this._dimensions
+    return this._dimensionsExpessions
```

Comment:
dimensions are not the same thing as dimension expressions, so there should be a different method to fetch dimension expressions. Also, shouldn't knowledge of dimension expressions be encapsulated within dimensions and exposed via accessors if needed?